src/ops.rs

```diff
@@ -594,56 +594,42 @@ fn op_open(
       open_options.read(true);
     }
     "r+" => {
-      if let Err(e) = state.check_write(&filename_str) {
-        return odd_future(e);
-      }
       open_options.read(true).write(true);
     }
     "w" => {
-      if let Err(e) = state.check_write(&filename_str) {
-        return odd_future(e);
-      }
       open_options.create(true).write(true).truncate(true);
     }
     "w+" => {
-      if let Err(e) = state.check_write(&filename_str) {
-        return odd_future(e);
-      }
       open_options
         .read(true)
         .create(true)
         .write(true)
         .truncate(true);
     }
     "a" => {
-      if let Err(e) = state.check_write(&filename_str) {
-        return odd_future(e);
-      }
       open_options.create(true).append(true);
     }
     "a+" => {
-      if let Err(e) = state.check_write(&filename_str) {
-        return odd_future(e);
-      }
       open_options.read(true).create(true).append(true);
     }
     "x" => {
-      if let Err(e) = state.check_write(&filename_str) {
-        return odd_future(e);
-      }
       open_options.create_new(true).write(true);
     }
     "x+" => {
-      if let Err(e) = state.check_write(&filename_str) {
-        return odd_future(e);
-      }
       open_options.create_new(true).read(true).write(true);
     }
     &_ => {
       panic!("Unknown file open mode.");
     }
   }
 
+  if mode != "r" {
```

Comment:
Better move to before the match to avoid file creation/truncation?

Reply:
The error is different for invalid mode. I'd like to keep the behavior (panic) unchanged when invalid mode is used.

Reply:
Valid point. However, I also believe that unexpected truncation of important files or creating files with file names containing sensitive information still feels like a very bad security breach...

Reply:
Yeah, but I guess it is a separate topic tracked by #720